### PR TITLE
Resolves GenericEnumVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the GenericEnumVisualizer into a component/system pair.

### TODO:
- [x] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Review by AJCM-git — CHANGES_REQUESTED** [review]
remove GenericEnumVisualizer and replace its usages with a gen vis

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed. It should be good to go whenever is convenient for you.

File: Resources/Prototypes/Entities/Structures/meat_spike.yml [hunk 1]
```diff
@@ -41,13 +41,12 @@
...
+  - type: GenericVisualizer
     visuals:
-      - type: GenericEnumVisualizer
-        key: enum.KitchenSpikeVisuals.Status
-        layer: 0
-        states:
-          enum.KitchenSpikeStatus.Empty: spike
-          enum.KitchenSpikeStatus.Bloody: spikebloody
+      enum.KitchenSpikeVisuals.Status:
+        base:
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/meat_spike.yml:47]: this has no base mapped

File: Resources/Prototypes/Entities/Objects/Fun/pai.yml [hunk 2]
```diff
@@ -20,8 +20,10 @@
     sprite: Objects/Fun/pai.rsi
     layers:
     - state: pai-base
+      map: ["base"]
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Fun/pai.yml:23]: is this used?

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Fun/pai.yml (hunk 2 above)
Comment (TemporalOroboros) [reply Resources/Prototypes/Entities/Objects/Fun/pai.yml:23]: No.

**Pull request merged**